go/README.md

```diff
@@ -0,0 +1,55 @@
+# Go Target for Snowball
+
+The initial implementation was built as a port of the Rust target.  The initial focus has been on getting it to function, and making it work correctly.  No attempt has been made to beautify the implementation, generate code, or address performance issues.
+
+## Usage
+
+Until this PR is accepted, don't use `go get` to get the runtime library in place.  Instead, clone the mschoch repo into `$GOPATH/src/github.com/snowballstem/snowball`.
+
+To generate Go source for a Snowball algorithm:
+```
+$ snowball path/to/algorithm.sbl -go -o algorithm
+```
+
+### Go specific options
+
+`-gop[ackage]` the package name used in the generated go file (defaults to `snowball`)
+
+`-gor[untime]` the import path used for the Go Snowball runtime (defaults to `github.com/snowballstem/snowball/go`)
+
+## Code Organization
+
+`compiler/generator_go.c` has the Go code generation logic
+
+`go/` contains the default Go Snowball runtime support
+
+`go/stemwords` contains the source for a Go version of the stemwords utility
+
+`go/algorithms` location where the makefile generated code will end up
+
+## Using the Generated Stemmers
+
+Assuming you generated a stemmer, put that code in a package which is imported by this code as `english`.
+
+```
+env := snowball.NewEnv("beautiful")
+english.Stem(env)
+fmt.Printf("stemmed word is: %s", env.Current())
+```
```

Comment:
That's what `SetCurrent()` ought to do - just changing the string but leaving the cursor pointing potentially past the end of it isn't a useful operation.

Looks like the Rust code you based this on doesn't handle this properly...